src/WebJobs.Extensions.DurableTask/Listener/TaskEntityShim.cs

```diff
@@ -741,11 +741,13 @@ private async Task ExecuteOutOfProcBatch()
         {
             Activity processEntityInvocationActivity = null;
             Activity callEntityActivity = null;
+            ActivityContext? parentTraceContext;
```

Comment:
I'm not quite understanding what we're changing in this file and need a little help. It looks like we're declaring two variables instead of one, but we're still only using one of those variables. Why was this change necessary?

Reply:
Yeah sorry, this is confusing. Basically, the problem is that I want this `parentTraceContext` to now be nullable so that I can assign to it from the `callEntityActivity?.Context` call. Previously, since I didn't need it to be nullable, I could just make it the `out` variable of the `ActivityContext.TryParse` call. But now, since it's nullable, I can't have it be the `out` variable of that call, so I have two variables - one is the `out` variable of the call and the other is this newly-nullable `parentTraceContext`. If the `TryParse` succeeds I just assign its result to `parentTraceContext`. Let me know if there's a better way to do this

Reply:
I'm a little lost on this one too TBH - the code compiles without any changes to TaskEntityShim, right?. Passing a non-nullable object (parentTraceContext in this case) to a nullable parameter is valid in .NET, but not the other way around. Unless you were running into some error at runtime

Reply:
Ah yeah that's not the issue, the issue is that I'm trying to _assign_ a nullable value (`callEntityActivity?.Context`) to a non-nullable parameter `parentTraceContext`. Previously, the call was instead `callEntityActivity.Context`, which had a non-nullable result, and could be assigned to the non-nullable `parentTraceContext`. (But this is precisely what is causing the null reference exception, since `callEntityActivity` can in fact be null). So I had to change `parentTraceContext` to be nullable

Reply:
Right yeah - I actually just enabled nullability for this solution and we have _a lot_ of dangerous null handling. 

I think you could do something like this here? Let me know if this makes sense - I'm not sure it's any more readable than what you have, though:
```csharp
parentTraceContext = callEntityActivity is not null ? callEntityActivity.Context : parentTraceContext;
```

Reply:
Or better, 
```csharp
parentTraceContext = callEntityActivity?.Context ?? parentTraceContext;
```

Reply:
The problem is that I actually want `parentTraceContext` to hold a null value in the case that `callEntityActivity` is null. (This is because I want to then pass a null value to `StartActivityForProcessingEntityInvocation` for the parent trace context in that case). So unfortunately I can't use your suggestion (although I agree it's more readable).